pkg/services/anonymous/anonimpl/impl.go

```diff
@@ -96,5 +120,26 @@ func (a *AnonDeviceService) TagDevice(ctx context.Context, httpReq *http.Request
 
 	a.localCache.SetDefault(key, struct{}{})
 
-	return a.remoteCache.Set(ctx, key, []byte(key), thirtyDays)
+	deviceJSON, err := json.Marshal(taggedDevice)
+	if err != nil {
+		return err
+	}
+
+	if err := a.remoteCache.Set(ctx, key, deviceJSON, thirtyDays); err != nil {
+		return err
+	}
+
+	untagKind := anonymous.AnonDevice
+	if kind == anonymous.AnonDevice {
+		untagKind = anonymous.AuthedDevice
+	}
+	if err := a.untagDevice(ctx, &Device{
```

Comment:
thinking about the case where we would add to the remoteCache and then the untagDevice failed.

is this an issue for us? 🤔